base_layer/transaction_components/src/key_manager/inner.rs

```diff
@@ -110,9 +110,10 @@ use crate::{
     MicroMinotari,
 };
 
+#[derive(Clone)]
 pub struct TransactionKeyManagerInner<TBackend> {
-    key_managers: HashMap<String, RwLock<TariKeyManager<KeyDigest>>>,
-    db: Option<TBackend>,
+    key_managers: HashMap<String, TariKeyManager<KeyDigest>>,
+    db: Option<Arc<RwLock<TBackend>>>,
```

Comment:
Since TBackend is an r2d2 SQLite WAL mode DB connection, it is already thread-safe and designed for concurrent access via its connection pool. We can safely remove the Arc<RwLock<...>> wrapper and use Option<TBackend> directly.
```suggestion
    db: Option<TBackend>,
```

Reply:
TBackend is a trait, not an object, it can be anything